### DISCUSSION TITLE
About the periodic boundary condition for non-orthogonal parallelepiped RVE model

### DISCUSSION THREAD

**Discussion by liuyii7** [neperfepx/neper#932] [Q&A] [answered]
I would like to consult that how to generate the periodic boundary condition for the parallelepiped RVE model in tessellation part? 

For example, I have the seed coordinates and their weight, and the Voronoi tessellation will be applied. However, the configuration of the RVE generated by seed coordinates and their weight is not the rectangular or the cube with the orthogonal edges, I want to generate the periodic condition for the paralleleipied RVE along each non-orthogonal direction.

**Comment by rquey** [comment, marked as answer]
You do not need straight edges etc. to do a simulation. The periodicity conditions apply to the (irregular shaped) periodic tessellations just as well. But if you want straight edges, etc., you may have a look at [Neper-mosaic](https://doi.org/10.1016/j.softx.2024.101912).

**Reply by liuyii7** [reply]
Thanks for your kindly reply! But actually it is not what I want to consult, I mean I want to generate the periodic condition along the axis that they are not perpendicular with each other. Is it possible in Neper?

**Reply by rquey** [reply]
Ah, sorry. I think it is not, but what would be the interest? Since the medium is periodic and fills the entire space by translation, in principle, it is always possible to consider a right-parallelepiped "unit cell", right?

**Reply by liuyii7** [reply]
I see, thanks again for your reply!

Yes, normally this is the case. However, I built a model that need to use the tessellation function for every iteration: for each iteration, I update the coordinates and the weight of each seed (they changes for each time, which means the RVE deformed in the space), then I use the information of seed to generate a new polycrystalline. 

The purpose for doing this is: under large deformation, some 'un-natural' grain boundary facets occurs. Therefore, I update the polycrystalline structure by using Voronoi tessellation for each step, and I could avoid the 'un-natural' grain boundary facets.

**Reply by liuyii7** [reply]
By the way, 'un-natural' grain boundary facets means the 'concave' polygon, and I want to keep all the grain boundary facets are 'convex' polygon during the entire simulation.

**Reply by rquey** [reply]
OK. Neper considers periodicity only along the reference directions right now. You could still load your updated "sheared" seed coordinates and weights, and build the tessellation in the equivalent "unsheared" domain. Neper would generate the tessellation in that "unsheared" domain, but this would be equivalent...
Hope this helps.

**Comment by liuyii7** [comment]
Ok, I got it. Really thanks for your guidance!

发送自 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Romain Quey ***@***.***>
发送时间: Saturday, October 26, 2024 12:33:07 AM
收件人: neperfepx/neper ***@***.***>
抄送: liuyii7 ***@***.***>; Author ***@***.***>
主题: Re: [neperfepx/neper] About the periodic boundary condition for non-orthogonal parallelepiped RVE model (Discussion #932)


OK. Neper considers periodicity only along the reference directions right now. You could still load your updated "sheared" seed coordinates and weights, and build the tessellation in the equivalent "unsheared" domain. Neper would generate the tessellation in that "unsheared" domain, but this would be equivalent...
Hope this helps.

―
